qiskit/extensions/standard/barrier.py

```diff
@@ -52,29 +51,32 @@ def qasm(self):
 
     def reapply(self, circ):
         """Reapply this gate to corresponding qubits in circ."""
-        self._modifiers(circ.barrier(*self.arg))
-
-
-def barrier(self, *tuples):
-    """Apply barrier to tuples (reg, idx)."""
-    tuples = list(tuples)
-    if not tuples:  # TODO: implement this for all single qubit gates
-        if isinstance(self, QuantumCircuit):
-            for register in self.regs.values():
-                if isinstance(register, QuantumRegister):
-                    tuples.append(register)
-    if not tuples:
-        raise ExtensionError("no arguments passed")
+        self._modifiers(circ._attach(Barrier(self.arg, self)))
+
+
+def barrier(self, *args):
+    """Apply barrier to circuit.
+    If args is None, applies to all the qbits.
+    If args is a QuantumRegister, applies to all the qbits in that register.
+    If args is a list of QuantumRegister, applies to all of them."""
+
```

Comment:
It looks like args can be a list of QuantumRegister and/or qubit, and the barrier will be applied to all arguments. If that's right, could you add that to the comment?

Reply:
Clarified in d3a9d32. Thanks!